core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1382,12 +1382,24 @@ object TestUtils extends Logging {
     records
   }
 
-  def createTransactionalProducer(transactionalId: String, servers: Seq[KafkaServer], batchSize: Int = 16384) = {
+  def consumeRemainingRecords[K, V](consumer: KafkaConsumer[K, V], timeout: Long): Seq[ConsumerRecord[K, V]] = {
```

Comment:
Nit: I think this should be `consumeRecordsFor` and `timeout` should be `duration`. Assuming I understood the intent (consume records until `timeout/duration` expires. And we should probably have scaladoc explaining this.

Reply:
fixed in #4146